### PR TITLE
[FIX] Added use real name check for attachment msg preview

### LINKED ISSUE #2896: LastMessage in Room list doesn't respect use real name setting.
### Description:
In room list screen, when an user sends an attachment, then the preview msg doesn't respects the use real name permission.

### Environment Information:

- Rocket.Chat Server Version: 3.12.0 - develop
- Rocket.Chat App Version: 4.14.1
- Device Name: Redmi k20
- OS Version: Android 10

### Steps to reproduce:

1.  Let any user send you an attachment
2.  Go to room list
3.  username is always displayed as - "username sent you an attachment"

### Expected behavior:
Names should respect use real name setting.

### Actual behavior:
doesn't respect use real name setting on room list screen for attachment

### Additional context:

image

image

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Proposed changes

Added use real name permission check for attachments when preview of last msg is shown on room list
## Issue(s)	

fixes #2896 
## How to test or reproduce

check issue.
## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

image

### CODE REVIEW

**Comment by yash-rajpal** [comment]
@diegolmello Please Review. :)

File: app/presentation/RoomItem/LastMessage.js [hunk 1]
```diff
@@ -26,7 +26,8 @@ const formatMsg = ({
 	const isLastMessageSentByMe = lastMessage.u.username === username;
 
 	if (!lastMessage.msg && lastMessage.attachments && Object.keys(lastMessage.attachments).length) {
-		const user = isLastMessageSentByMe ? I18n.t('You') : lastMessage.u.username;
+		const name = useRealName ? lastMessage.u.name : lastMessage.u.username;
```
Comment (diegolmello) [inline app/presentation/RoomItem/LastMessage.js:29]: Right below this, we have the same rule https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/2897/files#diff-477299f3632d70ed68c5a4dedd1c1432968769cc1d582fde95cc395682242a3eR43

Can you just move your declaration above on this scope and use it there as well, so we keep it DRY?

**Review by diegolmello — CHANGES_REQUESTED** [review]
@yash-rajpal Can you add stories for your PR?
We should make sure there's no regression.
Thanks!

image

File: app/presentation/RoomItem/LastMessage.js (hunk 1 above)
Comment (yash-rajpal) [reply app/presentation/RoomItem/LastMessage.js:29]: Sure. Will do :)

**Comment by yash-rajpal** [comment]
@diegolmello Done all requested changes and added stories.
Thanks. :)

**Pull request closed without merging**